storage_s3/indico_storage_s3/task.py

```diff
@@ -0,0 +1,40 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from datetime import datetime
+
+from celery.schedules import crontab
+from dateutil.relativedelta import relativedelta
+
+from indico.core.celery import celery
+from indico.core.storage.backend import get_storage
+
+
+@celery.periodic_task(run_every=crontab(minute=0, hour=1))
+def create_bucket():
+    storage = get_storage('s3')
+    bucket_name = storage.get_bucket_name(replace_placeholders=False)
+    now = datetime.now()
+    if all(x in bucket_name for x in ['<year>', '<week>']) and now.weekday() == 0:
```

Comment:
Isn't a simple `'<year>' in bucket_name and '<week>' in bucket_name` more readable? ;)

Reply:
Actually, looking at the checks below, what about extracting the set of placeholders (`re.findall` is probably the easiest option) and using set operations?

    if placeholders & {'<year>', '<month>'}:

This will be especially useful in the checks below